### PR TITLE
5481 asset import missing properties

### LINKED ISSUE #5481: iSC/Cold chain/GAPS: some asset attributes are not configured/missing from export/import functionality
## What went wrong? 😲

I was playing around with importing/exporting cold chain assets, and noticed some attributes were not included in the import template or when exporting a CSV file -> therefore the user would have to go through each asset manually to update the missing details. 

**The missing ones are:** 

- Store
- Category
- Type
- Warranty Start
- Warranty End
- Current Status
- Reason
- Needs Replacement (Y/N)
- Donor

**Enhancements:** 

- Can lifespan column be placed after Refrigerant Type(s) just as it appears in omSupply? Currently, it's the last column in the template/export file which does not make sense to me
- Temperature Monitoring Device appears twice in the file

Screenshot 2024-11-20 at 10 56 31 AM

- Catalogue Item code appears in export and is editable in import template, however does not appear in omSupply (front-end wise?). Should this info be visible to the users? Possibly needs @adamdewey's input

## Expected behaviour 🤔

If the field is editable/fillable by the user, it should be in the import template & also exported, otherwise the user will have to manually update what's missing

## How to Reproduce 🔨

Steps to reproduce the behaviour:

1. Go to cold chain -> assets
2. Create a new asset and fill out all the details (e.g. cold chain location, measurements, donor etc)
3. Export the list of assets - see some columns are missing, no way for the user to export some information
4. When uploading assets, download the template provided - notice the same columns are missing from the template
5. Inconsistency!

## Your environment 🌱

- Open mSupply Version: V2.4.0-RC1
- Legacy mSupply Central Server Version: V7.18.11

- Platform: Google Chrome (browser on Mac)

- Database type: SQLite3

### PR DESCRIPTION
**Pull request by lache-melvin**
Fixes #5481 

# 👩🏻‍💻 What does this PR do?

Adds columns to import/export of CCE Assets.

## 💌 Any notes for the reviewer?

Doesn't include the category/type/manufacturer (as this is derived from the `Catalogue item code`

Doesn't set Donor - names would be harder to navigate at this time

# 🧪 Testing

- [ ] Have some CCE assets
- [ ] Export assets
  - [ ] warranty dates, current status and whether it needs replacing should come through
  - [ ] Properties should not be duplicated
- [ ] Click import asset
  - [ ] Download the import CSV
  - [ ] Fill it out with some new assets (at least one functioning, one with a different status, one that needs replacing)
  - [ ] Import the CSV
  - [ ] Assets import correctly (dates and statuses mapped! - note that if you don't set a status, it will be created as functioning)
  - [ ] Import again to create errors (already exists)
  - [ ] Export the errors CSV
  - [ ] Columns all look correct here
- [ ] Compare all the CSV files, columns should now all be in the same order
- [ ] Check all works on both central server and remote site (central should include store column, remote should not)

# 📃 Documentation

- [ ] **Part of an epic**: documentation will be completed for the feature as a whole
- [x] **No documentation required**: no user facing changes or a bug fix which isn't a change in behaviour
- [ ] **These areas should be updated or checked**: 
  1.
  2.

### CODE REVIEW

File: client/packages/coldchain/src/Equipment/ImportAsset/ImportReviewDataTable.tsx [hunk 1]
```diff
@@ -29,24 +31,12 @@ export const ImportReviewDataTable: FC<ImportReviewDataTableProps> = ({
...
-      sortable: false,
-      label: 'label.asset-number',
-    },
-    {
-      key: 'catalogueItemCode',
-      width: 50,
-      sortable: false,
-      label: 'label.catalogue-item-code',
-    },
-  ];
```
Comment (lache-melvin) [inline client/packages/coldchain/src/Equipment/ImportAsset/ImportReviewDataTable.tsx:45]: Sorry, little bit of jumping around - didn't get why everywhere was a different order

File: client/packages/coldchain/src/Equipment/ImportAsset/ImportPanel.tsx [hunk 2]
```diff
@@ -4,6 +4,7 @@ import { styled, TabPanel, Box } from '@openmsupply-client/common';
 
 const StyledTabPanel = styled(TabPanel)({
   height: '100%',
+  width: '100%',
```
Comment (lache-melvin) [inline client/packages/coldchain/src/Equipment/ImportAsset/ImportPanel.tsx:7]: Review table wasn't scrolling/overflowing correctly with added columns

File: client/packages/coldchain/src/Equipment/ImportAsset/UploadTab.tsx [hunk 3]
```diff
@@ -210,13 +211,16 @@ export const EquipmentUploadTab: FC<ImportPanel & EquipmentUploadTabProps> = ({
...
         catalogueItemCode: '',
         store: undefined,
         notes: '',
         serialNumber: '',
-        installationDate: 'DD/MM/YYYY',
-        replacementDate: 'DD/MM/YYYY',
+        installationDate: t('label.date-format'),
+        replacementDate: t('label.date-format'),
+        warrantyStart: t('label.date-format'),
+        warrantyEnd: t('label.date-format'),
```
Comment (lache-melvin) [inline client/packages/coldchain/src/Equipment/ImportAsset/UploadTab.tsx:222]: realised these probably don't translate well!

File: client/packages/common/src/intl/locales/en/common.json [hunk 4]
```diff
@@ -1152,7 +1153,7 @@
   "messages.cannot-view-vaccine-card": "You cannot view this patient's Vaccination Card from this store, because they were enrolled in the {{ programName }} by another store",
   "messages.cant-delete-generic": "You cannot delete one or more of the selected items",
   "messages.cant-delete-requisitions": "Can only delete requisitions with a status of 'Draft'",
-  "messages.cant-download-android": "File cannot be download on this device",
+  "messages.cant-download-android": "File cannot be downloaded on this device",
```
Comment (lache-melvin) [inline client/packages/common/src/intl/locales/en/common.json:1156]: Side quest 😁

**Review by jmbrunskill — APPROVED** [review]
Happy to merge when you're ready...

I've found one bug of sorts though, spreadsheet program seems to love mangling date formats, so I think we should add a bit more validation around the dates.

Otherwise it's very easy to do this (i did it by accident)
image


image

image

I think we used to have it so that we had YYYY-MM-DD in the import. I think this is a much less ambiguous format for imports, I know it's not so common but I wonder if we should consider going back to it?

At least checking that we have 4 digits for the year would be good...

File: client/packages/coldchain/src/Equipment/ImportAsset/EquipmentImportModal.tsx [hunk 5]
```diff
@@ -70,12 +75,30 @@ export const toInsertEquipmentInput = (
...
   return {
     ...rest,
+    needsReplacement: !!row.needsReplacement,
     catalogueItemId,
     store: store ? { ...store, __typename: 'StoreNode', storeName: '' } : null,
     parsedProperties,
   };
 };
 
+export const toStatusTypeInput = (status: StatusType): AssetLogStatusInput => {
```
Comment (jmbrunskill) [inline client/packages/coldchain/src/Equipment/ImportAsset/EquipmentImportModal.tsx:85]: Can we add these to the docs somewhere, so people know what options they have for that column?

**Comment by noel-yeldos** [comment]
@jmbrunskill I tried re-importing the file and even in YYYY-MM-DD format it's still adding the leading zeroes. Is this what you what you were referring to?

Screenshot 2024-11-26 at 10 28 31 AM

FYI I had made a change to update the formatting in the import table to DD-MM-YYYY per issue #5447

**Comment by jmbrunskill** [comment]
> @jmbrunskill I tried re-importing the file and even in YYYY-MM-DD format it's still adding the leading zeroes. Is this what you what you were referring to?
> 
> Screenshot 2024-11-26 at 10 28 31 AM
> 
> FYI I had made a change to update the formatting in the import table to DD-MM-YYYY per issue #5447

Yes, clearly that doesn't check properly either :)

It's all good I think would just be good to add a check because we can't just re-import to fix and could be a systematic error for 1000s of assets...

**Review by noel-yeldos — APPROVED** [review]

**Pull request merged**